src/Compiler/Microsoft.CodeAnalysis.Razor/src/DefaultTypeNameFeature.cs

```diff
@@ -61,7 +61,12 @@ static ImmutableArray<string> ParseCore(TypeSyntax parsed)
                 return list;
             }
 
-            return ImmutableArray.Create(parsed.ToString());
+            if (parsed is IdentifierNameSyntax identifier)
```

Comment:
It doesn't feel particularly safe to me to assume this is an IdentifierNameSyntax. What about qualified names, or keywords?

Reply:
This should only return type parameters. Can they be anything else than IdentifierNameSyntax?

That makes me think of a different issue, though: `Dictionary<X, T>` (where `X` is a user-defined class in `global` namespace) doesn't work because `X` is parsed as type parameter as well. That's a different issue though, should be probably fixed separately. I will just add a test in this PR.

Reply:
Well, if there can tuple or array types, I would expect you could also have `A.B<T>` here.

Reply:
You're right. That will be ignored here and handled by the call `DescendantNodesAndSelf().OfType<TypeArgumentListSyntax>()` above.

Reply:
Let's put a comment on that then, so future readers aren't as confused as I was :).